sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -212,21 +212,21 @@ def load(*args, **kwargs) -> "AzureAppConfigurationProvider":
                         label,
                     )
                 else:
-                    _prekill(start_time)
+                    _min_uptime(start_time)
                     raise e
             except Exception as e:
-                _prekill(start_time)
+                _min_uptime(start_time)
                 raise e
     return provider
 
 
-def _prekill(start_time: datetime.datetime) -> None:
+def _min_uptime(start_time: datetime.datetime) -> None:
     # We want to make sure we are up a minimum amount of time before we kill the process. Otherwise, we could get stuck
     # in a quick restart loop.
     min_time = datetime.timedelta(seconds=min_uptime)
     current_time = datetime.datetime.now()
     if current_time - start_time < min_time:
```

Comment:
`current_time - start_time` will have changed between the time of the `if` check on line 228 and when it is re-evaluated on line 229 such that `(min_time - (current_time - start_time))` could actually be negative. 
Why don't we save the elapsed time in a variable to be safe. Granted, this will add on a bit of time to the actual wait but it would prevent any potential timing issues.

Thoughts?

Reply:
? start time and current time are set values, they will never update to be anything different.

Reply:
Right. I missed line 227. Feel free to ignore this.